File: config-resolver-json/src/main/java/org/godari/config/resolver/dto/OverrideRule.java
package org.godari.config.resolver.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.JsonNode;
import lombok.Data;

import java.util.Set;

@Data
public class OverrideRule {

    @JsonProperty("user-is-in-all-groups")
    private Set<String> userIsInAllGroups;

    @JsonProperty("user-is-in-any-group")
    private Set<String> userIsInAnyGroup;

    @JsonProperty("user-is-none-of-the-groups")
    private Set<String> userIsInNoneOfTheGroups;

    @JsonProperty("custom-expression")
    private String customExpression;

    @JsonProperty("override")
    private JsonNode override;
}

File: config-resolver-go/resolver/dto/override_rule.go
package dto

// OverrideRule defines the conditions and properties for override

type OverrideRule struct {
	UserIsInAllGroups       []string               `json:"user-is-in-all-groups"`
	UserIsInAnyGroup        []string               `json:"user-is-in-any-group"`
	UserIsInNoneOfTheGroups []string               `json:"user-is-none-of-the-groups"`
	CustomExpression        string                 `json:"custom-expression"`
	Override                map[string]interface{} `json:"override"`
}
